### DISCUSSION TITLE
How to make layers with {pmIgnore: true} snappable?

### DISCUSSION THREAD

**Discussion by aitanserikbayev** [geoman-io/leaflet-geoman#848] [Questions & Help]
I used the older version of leaflet-geoman in past and pmIgnore layers were snappable by default. Now they aren't. Is it possible to do them snappable but not editable and removable?

**Comment by Falke-Design** [comment]
Currently it is not really easy to solve, I think we should make this easier.

You have to add the layers they should snap manually on every drawstart or enabling:

```

var rect = L.rectangle(map.getBounds().pad(-0.3),{pmIgnore: true}).addTo(map);
var rect2 = L.rectangle(map.getBounds().pad(-0.6),{pmIgnore: true, color: 'red'}).addTo(map);


map.on('pm:drawstart',(e)=>{
  console.log(e);
  var pmDraw = map.pm.Draw[e.shape];
  // pmDraw._otherSnapLayers is cleared every time drawing is started 
  pmDraw._otherSnapLayers.push(rect); // Add all layers they should be snappable
  pmDraw._otherSnapLayers.push(rect2); 
  pmDraw._createSnapList();
});

map.on('pm:create',(e)=>{
  e.layer.on('pm:enable',(x)=>{
    console.log(x);
    var pm = x.layer.pm;
    // pm._otherSnapLayers is NEVER cleared, so we need to check if the layers to snap are already in the array
    pm._otherSnapLayers = pm._otherSnapLayers || []; // create array if it not exists
    if(pm._otherSnapLayers.indexOf(rect) === -1) { // check if layer is already in the array
      pm._otherSnapLayers.push(rect); // Add all layers they should be snappable
    }
    if(pm._otherSnapLayers.indexOf(rect2) === -1) {
      pm._otherSnapLayers.push(rect2); // Add all layers they should be snappable
    }
    console.log(pm._otherSnapLayers);
    pm._createSnapList();
  });
});
```

-----

> {pmIgnore: true, snapIgnore: false} should work

In the latest release we "fixed" that when a layer has `pmIgnore: true` that it is completly ignored from Geoman, so the option `snapIgnore: false` change nothing.
https://github.com/geoman-io/leaflet-geoman/blob/25911084c4c93598b5d936bd85392432efc65532/src/js/Mixins/Snapping.js#L171-L184

**Comment by henrikmaa** [comment]
I think if pmIgnore: true it should still not snap by default, but snapIgnore: false or snappable: true, should allow.

This became an issue since #677 (before you had too { pmIgnore: true, snapIgnore:true }) to ignore snap)

**Comment by Falke-Design** [comment]
Yes I see it like you. I will talk with @codeofsumit and then implement it.
